native/kotlin/api/kotlin/src/main/kotlin/rs/wordpress/api/kotlin/WpLoginClient.kt

```diff
@@ -19,4 +19,29 @@ class WpLoginClient(
     suspend fun apiDiscovery(siteUrl: String): AutoDiscoveryUniffiResult = withContext(dispatcher) {
         internalClient.apiDiscovery(siteUrl)
     }
+
+    suspend fun loginUrl(siteUrl: String): String = withContext(dispatcher) {
```

Comment:
This implementation feels really off to me. (to be clear, not because of how you did it)

 I think we should add a helper function to Rust for it that returns a `Result` so that when we use it we either get the successful url or the most relevant error.

If you agree, I can open a PR about this targeting `trunk`, so you can rebase and update this function. We also probably should return a [Kotlin Result](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-result/) from this function.

Reply:
I'm fine to move it into Rust – it needed to be implemented separately for Swift as well.